src/js/main.js

```diff
@@ -177,21 +203,37 @@ Site.Shapes = {
   timer: null,
   interval: 500,
   animating: false,
+  atLimit: false, // is the map at a limit?
 
   init: function() {
     var _this = this;
 
     _this.showPattern();
 
-    var map = document.getElementById('map');
+    Site.mapElement.addEventListener('startpan', _this.startAnimation.bind(_this));
 
-    map.addEventListener('startpan', function() {
-      _this.startAnimation();
-    });
+    Site.mapElement.addEventListener('stoppan', _this.stopAnimation.bind(_this));
 
-    map.addEventListener('stoppan', function() {
-      _this.stopAnimation();
-    });
+    Site.mapElement.addEventListener('movemap', _this.checkMapPosition.bind(_this));
+
+  },
+
+  // Check if the map position is at a limit
+  checkMapPosition: function(event) {
+    var _this = this;
+
+    var x = event.detail.map.position.x;
+    var y = event.detail.map.position.y;
+
+    // Check for left limit
+    if (x >= 0 ||
+      x <= Site.window.width * -2 || // Check for right limit
+      y >= 0 || // Check for upper limit
+      y <= Site.window.height * -2) { // Check for bottom limit
+      _this.atLimit = true;
+    } else {
```

Comment:
both things are easy to do but centers would be hard to hit so they are kinda useless